app/server/methods/residents.js

```diff
@@ -87,5 +87,59 @@ Meteor.methods({
 
     // Sum of duration for particular resident & activity
     return residentActivities.reduce((prevResult, document) => prevResult + document.duration, 0)
+  },
+  residentNamesGroupedtByHomes(){
+
+    // Get list of homes, sorted alphabetically
+    const homes = Homes.find({}, { sort: { name: 1 } }).fetch();
+
+    // Create an array residents grouped by home
+    const residentsSelectOptions = _.map(homes, function(home) {
+      // Get home ID
+      const homeId = home._id;
+
+      // do not show departed residents
+      const departed = {
+        moveOut: {
+          $exists: false
+        }
+      };
+
+      // Sort by first name in alphabetical order
+      const sort = { firstName: 1 };
+
+      // Get a list of residents for current home
+      const homeResidents = Residencies.find(
+        { homeId, ...departed },
+        { sort }
+      ).fetch().map(function(resident){
+        const residentId = resident.residentId
+        const residentDet = Residents.findOne(residentId)
+         
+
+          let fullName = residentDet.fullName()
```

Comment:
Always default to `const`. `let` is only necessary when the *reference* will change. Since you are not reassigning `fullName`, it should be defined with `const`.